src/transformers/modeling_utils.py

```diff
@@ -495,6 +495,18 @@ def tie_weights(self):
                 self = getattr(self, self.base_model_prefix)
             self._tie_encoder_decoder_weights(self.encoder, self.decoder, self.base_model_prefix)
 
+    @staticmethod
+    def _tie_encoder_decoder_word_embeddings(encoder: nn.Module, decoder: nn.Module):
+        encoder_embeddings = encoder.get_input_embeddings()
+        decoder_embeddings = decoder.get_input_embeddings()
+        if encoder_embeddings is not None and decoder_embeddings is not None:
+            if decoder_embeddings.weight.size() == encoder_embeddings.weight.size():
+                # the priority is given to encoder embeddings, because e.g. in Longformer2Roberta case Longformer
+                # embeddings are additionally pretrained for MLM task
+                decoder_embeddings.weight = encoder_embeddings.weight
+                # tie input and output embeddings of decoder
+                decoder.tie_weights()
```

Comment:
Can we remove the `decoder.tie_weights()` function? -> This is not really the responsibility of the `_tie_encoder_decoder_word_embeddings()` function, but of the decoder himself.

Reply:
Before we tie encoder-decoder embedding matrices, decoder output matrix is tied to decoder input matrix. If I remove this line decoder output matrix will point to the old decoder embedding matrix.

Reply:
I see! Could we solve this by setting `encoder_embeddings.weight = decoder_embeddings` ? This would be the cleaner way anyways because we also use the direction encoder ties from decoder in `_tie_encoder_decoder_weights`

Reply:
Yeah, but as I commented a couple of lines above, in case of Longformer2Roberta, Longformer embeddings are additionally trained so I give the priority to encoder embeddings assuming they are usually more important.

Reply:
Yeah, but as I commented a couple of lines above, in case of Longformer2Roberta, Longformer embeddings are additionally trained so I give the priority to encoder embeddings assuming they are usually more important. And `_tie_encoder_decoder_weights ` is not always called, e.g. if model classes are different.

Reply:
True it's actually not trivilal to decide which word embeddings to use in case they are different for the encoder and decoder...

The general philosophy of the library is to not add too much "magic" to the functions that could result in weird unexpected behavior, so thinking a bit more about it, it might be better to actually not tie the word embeddings at all if they are different in value => we could add a check with `if not torch.allclose(encoder_embeddings.weight, decoder_embeddings.weigth, atol=1e-3)`: 
that prevents the word embeddings from being tied and prints a very explicit `logger.warn` message stating the encoder and decoder word embeddings are not the same and that in this case the user should tie the embeddings manually with `model.encoder.word_embeddings = model.decoder.word_embeddings` or the other way around. 

What do you think @alexyalunin?

Reply:
If embedding matrices are the same, in 99% of the cases the models are the same.
Now it is hard for me to imagine when the user may want to tie embeddings but not entire models (according to paper tying entire models is the best strategy). 

I actually made a false assumption that tying embeddings between models like longformer and roberta is always a good idea. Following the principle of least surprise and the philosophy of ''huggingface' library, it actually seems better to leave the option `model.encoder.word_embeddings = model.decoder.word_embeddings` to the user and not tie embeddings by default.

Reply:
@patrickvonplaten what do you think?

Reply:
I do think that we should still add this function. For models lke `Bert2Bert` I think this function shuold tie the embeddings by default. However for models such as Roberta2Longformer IMO it should not. 

Think there are many cases where an `untied` Bert2Bert performs better than a `tied` Bert2Bert => so it  would be nice to have this functonality :-)